src/main/java/org/mule/service/scheduler/internal/threads/SchedulerThreadPools.java

```diff
@@ -87,9 +93,17 @@
   private ScheduledThreadPoolExecutor scheduledExecutor;
   private org.quartz.Scheduler quartzScheduler;
 
-  private List<Scheduler> activeSchedulers = synchronizedList(new ArrayList<>());
+  private ReadWriteLock activeSchedulersLock = new ReentrantReadWriteLock();
+  private Lock activeSchedulersReadLock = activeSchedulersLock.readLock();
+  private Lock activeSchedulersWriteLock = activeSchedulersLock.writeLock();
+
+  private List<Scheduler> activeCpuLightSchedulers = new ArrayList<>();
+  private List<Scheduler> activeIoSchedulers = new ArrayList<>();
+  private List<Scheduler> activeCpuIntensiveSchedulers = new ArrayList<>();
+  private List<Scheduler> activeCustomSchedulers = new ArrayList<>();
 
-  public SchedulerThreadPools(String name, SchedulerPoolsConfig threadPoolsConfig) {
+  public SchedulerThreadPools(String name, SchedulerPoolsConfig threadPoolsConfig, boolean logRejectionDetails) {
```

Comment:
if flag needed, can't you just refer to static value of system property once read in this code?